dask_expr/io/io.py

```diff
@@ -173,9 +173,12 @@ def _task(self, index: int):
 
     @functools.cached_property
     def _fusion_buckets(self):
-        step = math.ceil(1 / self.operand("expr")._fusion_compression_factor)
         partitions = self.operand("expr")._partitions
         npartitions = len(partitions)
+
+        step = math.ceil(1 / self.operand("expr")._fusion_compression_factor)
+        step = min(step, math.ceil(math.sqrt(npartitions)), 100)
```

Comment:
My guess is that 100 is too high here, and that if we are genuinely collecting from 100 different files in one task that we'll want more parallelism.  I would probably aim to a maximum task-reduction factor of 10 or 20 or so.  

I'm happy to defer this discussion to later though with some benchmarking if it comes up.

Reply:
Yeah I'd rather do some benchmarking before lowering this too drastically, 100 also means that we have at least 10k files, which is probably a large dataset or many many small files